src/integrationtests/java/com/aws/greengrass/integrationtests/SyncTest.java

```diff
@@ -112,9 +116,11 @@
 
     private Supplier<Optional<SyncInformation>> syncInfo;
     private Supplier<Optional<ShadowDocument>> localShadow;
+    private final Lock sequential = new ReentrantLock();
 
     @BeforeEach
     void setup() {
+        sequential.lock();
```

Comment:
why do you need a lock in setup / cleanup?

Reply:
This was one of the options I had tried to make the tests less flaky. I thought I had removed it. But removed it now.